code/kit_version.R

```diff
@@ -0,0 +1,107 @@
+##	coefficients on temperature term (hypothetical linear response)
+	beta = 0.25
+##	coefficients on long-run temperature interaction term
+	gamma = 0.02
+##	constant term
+	alpha = 12000
+##	difference in pixel temperature and polynomial means
+	pixel_meantemp_diff = 20
+
+# REAL COUNTRY-LEVEL TEMP DATA
+
+##	importing
+	real_temps <- read_dta("Europe_temps.dta")
+	d <- real_temps
+##	create country ID
+	d$group <- d %>% group_indices(country) 
+##	create count of countries
+	country_count = length(unique(d$group))
+##	create mean-temp-by-country variable & deviations variable
+	d = d %>%
+	  group_by(country) %>%
+	  mutate(avgtemp1 = mean(temp1),
+	  		deviation = temp1 - avgtemp1) %>%
+	  ungroup()
+
+# FAKE DATA
+
+##	Generate positively and negatively correlated pixel temperature
+	d = d %>%
+	  mutate(temp1_pospixel1 = avgtemp1 + ((pixel_meantemp_diff)/2)*365.25 + deviation,
+	  		temp1_pospixel2 = avgtemp1 - ((pixel_meantemp_diff)/2)*365.25 + deviation,
+			temp1_negpixel1 = avgtemp1 + ((pixel_meantemp_diff)/2)*365.25 + 4*deviation,
+			temp1_negpixel2 = avgtemp1 - ((pixel_meantemp_diff)/2)*365.25 - 2*deviation)
+
+##	Generate long-run average temperature
+	d = d %>%
+	  mutate(lrtemp_country = avgtemp1/365.25)
+
+##	Generate long-run average temperature by pixel pair
+	d = d %>%
+	  group_by(country) %>%
+	  mutate(lrtemp_pospixel1=mean(temp1_pospixel1)/365.25,
+	  		lrtemp_pospixel2=mean(temp1_pospixel2)/365.25,
+	  		lrtemp_negpixel1=mean(temp1_negpixel1)/365.25,
+	  		lrtemp_negpixel2=mean(temp1_negpixel2)/365.25) %>%
+	  ungroup()
+
+	  ### THIS IS STRANGE: IN STATA THESE ARE NOT 0, IN R THEY ARE (significant digits?)
+	  ### A sanity check - do these vars = 0?
+	  d_check = d %>%
+	  mutate(checkpos = (temp1_pospixel1 + temp1_pospixel2)*0.5 - temp1,
+	  		checkneg = (temp1_negpixel1 + temp1_negpixel2)*0.5 - temp1)
+	  View(d_check)
+
+## 	Generate temperature * long-run average temperature, country-level values multiplied
+	d = d %>%
+	  mutate(interaction_country = temp1 * lrtemp_country)
+	  	
+##	Generate temperature * long-run average temperature, 
+##	pixel-level values multiplied and then aggregated to country
+	d = d %>%
+		  mutate(interaction_pospixel = (temp1_pospixel1*lrtemp_pospixel1 + temp1_pospixel2*lrtemp_pospixel2)*0.5,
+		  		interaction_negpixel = (temp1_negpixel1*lrtemp_negpixel1 + temp1_negpixel2*lrtemp_negpixel2)*0.5)
+
+# GENERATING FAKE OUTCOMES
+
+## 	Function to generate fake outcome data in the case of positively & negatively correlated pixel temperatures
+gen_outcomes <- function(d,draws,beta,a,gamma,b_pos,b_neg,alpha,sd) {
+  for(i in 1:draws) {
+    print(i)
+    a = as.character(a)
+    b_pos = as.character(b_pos)
+    b_neg = as.character(b_neg)
+    ypos = as.character(paste0("outcome_pos_", i))
+    yneg = as.character(paste0("outcome_neg_", i))
+    d[ypos] = beta*d[a] + gamma*d[b_pos] + alpha + sd*rnorm(1)
+    d[yneg] = beta*d[a] + gamma*d[b_neg] + alpha + sd*rnorm(1)
+  }
+   return(d)
+}
```

Comment:
This is fine, but you could eliminate this loop and replace it with a call to `lapply` or `mclapply`. It would be a bit involved to re-write, so let's discuss on the phone. Also, if you're passing strings to the function, it shouldn't be necessary to have the `as.character` calls.

Reply:
Also: you can use `glue()` (from the `glue` package) instead of `paste0()`: `paste0("outcome_pos_", i)` becomes `glue("outcome_pos_{i}")`